### PR TITLE
Fix broken doc tests in consumer

### PR DESCRIPTION
**Pull request by nicholastmosher**
There have been some API changes that broke some doc tests. For some reason it was not picked up by CI, I will open an issue to make sure doctests are run in CI. This fixes broken doc tests from consumer. This can be verified by running `cargo test --doc`.

### CODE REVIEW

File: src/client/src/consumer.rs [hunk 1]
```diff
@@ -76,15 +76,12 @@ impl PartitionConsumer {
...
+    /// # async fn example(consumer: &PartitionConsumer) -> Result<(), FluvioError> {
     /// let response = consumer.fetch(Offset::beginning()).await?;
     /// for batch in response.records.batches {
     ///     for record in batch.records {
-    ///         if let Some(record) = record.value.inner_value() {
-    ///             let string = String::from_utf8(record)
-    ///                 .expect("record should be a string");
-    ///             println!("Got record: {}", string);
-    ///         }
+    ///         let string = String::from_utf8_lossy(record.value.as_ref());
```
Comment (sehz) [inline src/client/src/consumer.rs:83]: there is new `to_string_lossy` in the record

Comment (nicholastmosher) [reply src/client/src/consumer.rs:83]: Hmm that's right... I wish I had seen that before we added it, I think it was not a good idea to add `to_string_lossy` to our record type. Is this a blocker in your eyes or could we merge it like this anyways?

Comment (sehz) [reply src/client/src/consumer.rs:83]: @simlay opinion?

Comment (simlay) [reply src/client/src/consumer.rs:83]: I'm fine with it as is and we can remove to_string_lossy

**Review by sehz — APPROVED** [review]

**Comment by sehz** [comment]
that's fine. I will merge it.

**Pull request merged**